CI stuck failing due to trivy scan detecting CVEs

### Bug Description

The CI for PRs is failing because on the Scan image job, causing any other job in the sequence to fail immediately. The failure is due to the `trivy` scan returning two high vulnerabilities (the default behaviour when vulnerabilities are found is to fail the job), which is scanning `ghcr.io/canonical/temporal-worker:e2524cf20e20b00369a15052e849f595ac2331cd-_1.0_amd64`.

NOTE it's been a while since this image was last updated and published. Potentially the Workflows Charmers team will not publish it any more in the future.

### To Reproduce

Run the CI to execute tests, wait for the `Scan images` job to complete and fail.

### Environment

This repo's CI environment.

### Relevant log output

```shell
Python (python-pkg)
===================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬───────────────┬──────────┬────────┬───────────────────┬────────────────────────┬─────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability │ Severity │ Status │ Installed Version │     Fixed Version      │                          Title                          │
├─────────────────────┼───────────────┼──────────┼────────┼───────────────────┼────────────────────────┼─────────────────────────────────────────────────────────┤
│ protobuf (METADATA) │ CVE-2025-4565 │ HIGH     │ fixed  │ 3.20.3            │ 4.25.8, 5.29.5, 6.31.1 │ python-protobuf: Unbounded recursion in Python Protobuf │
│                     │               │          │        │                   │                        │ https://avd.aquasec.com/nvd/cve-2025-4565               │
└─────────────────────┴───────────────┴──────────┴────────┴───────────────────┴────────────────────────┴─────────────────────────────────────────────────────────┘

usr/bin/pebble (gobinary)
=========================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                        Title                         │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2025-22874 │ HIGH     │ fixed  │ v1.24.2           │ 1.24.4        │ crypto/x509: Usage of ExtKeyUsageAny disables policy │
│         │                │          │        │                   │               │ validation in crypto/x509                            │
│         │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2025-22874           │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────┘
Error: Process completed with exit code 1.
```

### Additional context

[Logs](https://github.com/canonical/temporal-worker-k8s-operator/actions/runs/15864922634/job/44730331002?pr=47#step:8:437)

--- a/.github/workflows/integration_test.yaml
+++ b/.github/workflows/integration_test.yaml
@@ -12,9 +12,12 @@ jobs:
   integration-tests:
     uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
     secrets: inherit
+    env:
+      TRIVY_EXIT_CODE: '0'
     with:
       channel: 1.25-strict/stable
       modules: '["test_charm.py", "test_scaling.py", "test_vault.py", "test_postgres_db.py"]'
       juju-channel: 3.6/stable
       self-hosted-runner: false
       microk8s-addons: "dns ingress rbac storage metallb:10.15.119.2-10.15.119.4 registry"
+      upload-image: "artifact"

--- a/.github/workflows/integration_test.yaml
+++ b/.github/workflows/integration_test.yaml
@@ -10,7 +10,7 @@ on:
 
 jobs:
   integration-tests:
-    uses: dnplas/operator-workflows/.github/workflows/integration_test.yaml@dnplas-trivy-exit-code-as-input
+    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
     secrets: inherit
     with:
       channel: 1.25-strict/stable

--- a/.github/workflows/integration_test.yaml
+++ b/.github/workflows/integration_test.yaml
@@ -10,10 +10,8 @@ on:
 
 jobs:
   integration-tests:
-    uses: canonical/operator-workflows/.github/workflows/integration_test.yaml@main
+    uses: dnplas/operator-workflows/.github/workflows/integration_test.yaml@dnplas-trivy-exit-code-as-input
     secrets: inherit
-    env:
-      TRIVY_EXIT_CODE: '0'
     with:
       channel: 1.25-strict/stable
       modules: '["test_charm.py", "test_scaling.py", "test_vault.py", "test_postgres_db.py"]'
@@ -21,3 +19,4 @@ jobs:
       self-hosted-runner: false
       microk8s-addons: "dns ingress rbac storage metallb:10.15.119.2-10.15.119.4 registry"
       upload-image: "artifact"
+      trivy-exit-code: '0'

